airbyte-server/src/main/java/io/airbyte/server/ConfigDumpImport.java

```diff
@@ -0,0 +1,317 @@
+      TrackingClientSingleton.get().identify();
+      // report that the previous customer id is now superseded by the imported one.
+      previousCustomerIdOptional.ifPresent(
+          previousCustomerId -> TrackingClientSingleton.get().alias(previousCustomerId.toString()));
+    } catch (IOException | JsonValidationException | RuntimeException e) {
+      LOGGER.error("Import failed", e);
+      result = new ImportRead().status(StatusEnum.FAILED).reason(e.getMessage());
+    }
+
+    return result;
+  }
+
+  public void copySeed(Path targetPath) {
+    try {
+      FileUtils.copyDirectory(latestSeed.toFile(), targetPath.toFile());
+    } catch (IOException e) {
+      throw new RuntimeException(e);
+    }
+  }
+
+  private void checkImport(Path tempFolder) throws IOException, JsonValidationException {
+    final Path versionFile = tempFolder.resolve(VERSION_FILE_NAME);
+    final String importVersion = Files.readString(versionFile, Charset.defaultCharset())
+        .replace("\n", "").strip();
+    LOGGER.info(String.format("Checking Airbyte Version to import %s", importVersion));
+    if (!AirbyteVersion.isCompatible(targetVersion, importVersion)) {
+      throw new IOException(String
+          .format("Imported VERSION (%s) is incompatible with current Airbyte version (%s).\n" +
+              "Please upgrade your Airbyte Archive, see more at https://docs.airbyte.io/tutorials/upgrading-airbyte\n",
+              importVersion, targetVersion));
+    }
+    checkDBVersion(targetVersion);
+    importConfigsFromArchive(tempFolder, true);
+  }
+
+  // Config
+  private List<String> listDirectories(Path sourceRoot) throws IOException {
+    try (Stream<Path> files = Files.list(sourceRoot.resolve(CONFIG_FOLDER_NAME))) {
+      return files.map(c -> c.getFileName().toString())
+          .collect(Collectors.toList());
+    }
+  }
+
+  private <T> void importConfigsFromArchive(final Path sourceRoot, final boolean dryRun) throws IOException, JsonValidationException {
+    List<String> sourceDefinitionsToMigrate = new ArrayList<>();
+    List<String> destinationDefinitionsToMigrate = new ArrayList<>();
+    final boolean[] sourceProcessed = {false};
+    final boolean[] destinationProcessed = {false};
+    List<String> directories = listDirectories(sourceRoot);
+    Collections.sort(directories);
+    LinkedHashMap<ConfigSchema, Stream<T>> data = new LinkedHashMap<>();
+    for (String directory : directories) {
+      ConfigSchema configSchema = ConfigSchema.valueOf(directory.replace(".yaml", ""));
+      Stream<T> configs = readConfigsFromArchive(sourceRoot, configSchema);
+      configs = streamWithAdditionalOperation(sourceDefinitionsToMigrate, destinationDefinitionsToMigrate,
+          sourceProcessed,
+          destinationProcessed, configSchema, configs);
+      data.put(configSchema, configs);
+    }
+    configRepository.atomicImportOfConfigs(data, dryRun, dryRun ? path -> {} : (this::copySeed));
```

Comment:
the `copySeed` thing here is weird. why does the database need to know about this. can't we just iterate over the incoming records from the migration and the latest seed records and do the merge before we talk to the database? then we just pass only the already merged data to the database.

Reply:
I am thinking about how to address this.